google/resource_sql_database.go

```diff
@@ -112,10 +133,57 @@ func resourceSqlDatabaseRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("name", db.Name)
 	d.Set("self_link", db.SelfLink)
 	d.SetId(instance_name + ":" + database_name)
+	d.Set("charset", db.Charset)
+	d.Set("collation", db.Collation)
 
 	return nil
 }
 
+func resourceSqlDatabaseUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	database_name := d.Get("name").(string)
+	instance_name := d.Get("instance").(string)
+
+	db := &sqladmin.Database{
+		Name:     database_name,
+		Instance: instance_name,
+	}
+
+	if v, ok := d.GetOk("charset"); ok {
```

Comment:
Same as above -  we will always have a value present, so we can use `d.Get` instead of `d.GetOk`, and inline these into `db`